src/components/ExpenseUseForm.js

```diff
@@ -0,0 +1,458 @@
+import styled from 'styled-components';
+import { useForm } from 'react-hook-form';
+import { useNavigate } from 'react-router-dom';
+import { v4 as uuidv4 } from 'uuid';
+
+export default function ExpenseUseForm({
+  members,
+  expenses,
+  setExpenses,
+  editExpense,
+  buttonContent,
+}) {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    defaultValues: editExpense
+      ? {
+          id: editExpense.id,
+          title: editExpense.title,
+          description: editExpense.description,
+          amount: editExpense.amount,
+          date: editExpense.date,
+          paidBy: editExpense.paidBy,
+          paidFor: editExpense.paidFor,
+        }
+      : { date: new Date().toISOString().split('T')[0] },
+  });
+  const navigate = useNavigate();
+  const onSubmit = data => {
+    console.log(data);
+    console.log(expenses);
+    console.log(data.id);
+    console.log(expenses.findIndex(expense => expense.id === data.id));
+    console.log(
+      expenses.slice(
+        0,
+        expenses.findIndex(expense => expense.id === data.id)
+      )
+    );
+
+    data.id
+      ? setExpenses([
+          ...expenses.slice(
+            0,
+            expenses.findIndex(expense => expense.id === data.id)
+          ),
+          { ...data },
+          ...expenses.slice(
+            expenses.findIndex(expense => expense.id === data.id) + 1
+          ),
+        ])
+      : // Lösung: setExpenses expeneses slicen vorher, dann element bearbeiten dann slicen und alles spreaden
+
+        setExpenses(prevExpenses => [
+          ...prevExpenses,
+          { ...data, id: uuidv4() },
+        ]);
+    navigate(`/`);
+  };
+
+  if (members && members.length > 0) {
+    return (
+      <Wrapper>
+        <Form onSubmit={handleSubmit(onSubmit)} autocomplete="off">
+          <Error>{errors.title && <p>{errors.title.message}</p>}</Error>
+          <Error>{errors.amount && <p>{errors.amount.message}</p>}</Error>
+          <Error>{errors.date && <p>{errors.date.message}</p>}</Error>
+          <Error>{errors.paidBy && <p>{errors.paidBy.message}</p>}</Error>
+          <Error>{errors.paidFor && <p>{errors.paidFor.message}</p>}</Error>
+
+          <InputContainer>
+            <StyledInput
+              type="text"
+              id="title"
+              placeholder=" "
```

Comment:
If you don't use a placeholder you don't need to set the attribute. For all inputs in this form. 🙂

Reply:
Actually necessary as I use the 'placeholder-shown' pseudo-class to move the labels as soon as you focus the input. (Line 284)